src/views/sources/viewSourcesList.tsx

```diff
@@ -544,39 +443,60 @@ const SourcesListView: React.FunctionComponent = () => {
       )}
-
-      {!!pendingDeleteSource && (
-        <Modal
-          variant={ModalVariant.small}
-          title={t('form-dialog.confirmation', { context: 'title_delete-source' })}
-          isOpen={!!pendingDeleteSource}
-          onClose={() => setPendingDeleteSource(undefined)}
-          actions={[
-            <Button key="confirm" variant="danger" onClick={() => onDeleteSource()}>
-              {t('form-dialog.label', { context: 'delete' })}
-            </Button>,
-            <Button key="cancel" variant="link" onClick={() => setPendingDeleteSource(undefined)}>
-              {t('form-dialog.label', { context: 'cancel' })}
-            </Button>
-          ]}
-        >
-          Are you sure you want to delete the source &quot;
-          {pendingDeleteSource.name}&quot;
-        </Modal>
-      )}
-
-      {sourceBeingEdited && (
-        <AddSourceModal
-          source={sourceBeingEdited}
-          type={sourceBeingEdited.source_type}
-          onClose={() => setSourceBeingEdited(undefined)}
-          onSubmit={onSubmitEditedSource}
-        />
-      )}
-      {addSourceModal && (
-        <AddSourceModal type={addSourceModal} onClose={() => setAddSourceModal(undefined)} onSubmit={onAddSource} />
-      )}
-
+      <Modal
+        variant={ModalVariant.small}
+        title={t('form-dialog.confirmation', { context: 'title_delete-source' })}
+        isOpen={pendingDeleteSource !== undefined}
+        onClose={() => setPendingDeleteSource(undefined)}
+        actions={[
+          <Button
+            key="confirm"
+            variant="danger"
+            onClick={() => {
+              if (pendingDeleteSource) {
+                deleteSources(pendingDeleteSource).finally(() => {
+                  setPendingDeleteSource(undefined);
+                  onRefresh();
+                });
+              }
+            }}
+          >
+            {t('table.label', { context: 'delete' })}
+          </Button>,
+          <Button key="cancel" variant="link" onClick={() => setPendingDeleteSource(undefined)}>
+            {t('form-dialog.label', { context: 'cancel' })}
+          </Button>
+        ]}
+      >
+        {t('form-dialog.confirmation_heading', {
+          context: 'delete-source',
+          name: pendingDeleteSource?.name
+        })}
+      </Modal>
+      {/* TODO: Simplify add/edit sources functionality * check PR #380 for details */}
+      <AddSourceModal
+        isOpen={!!sourceBeingEdited}
+        source={sourceBeingEdited}
+        sourceType={sourceBeingEdited?.source_type ?? ''}
+        onClose={() => setSourceBeingEdited(undefined)}
+        onSubmit={(payload: SourceType) =>
+          editSources(payload).finally(() => {
+            queryClient.invalidateQueries({ queryKey: [SOURCES_LIST_QUERY] });
+            setSourceBeingEdited(undefined);
+          })
+        }
+      />
+      <AddSourceModal
+        isOpen={!!addSourceModal}
+        sourceType={addSourceModal ?? ''}
```

Comment:
required: remove the `?? ''` we're going to allow undefined in the props of the component